server/src/main/java/org/candlepin/resource/dto/ActivationKeyData.java

```diff
@@ -0,0 +1,325 @@
+        this.name = name;
+        return this;
+    }
+
+    public String getName() {
+        return this.name;
+    }
+
+    public ActivationKeyData setDescription(String description) {
+        this.description = description;
+        return this;
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    public ActivationKeyData setOwner(Owner owner) {
+        this.owner = owner;
+        return this;
+    }
+
+    public Owner getOwner() {
+        return this.owner;
+    }
+
+    public boolean addPool(Pool pool, Long quantity) {
+        String poolId = pool.getId();
+        this.pools.put(poolId, new AKPoolData(poolId, quantity));
+
+        return true;
+    }
+
+    public boolean removePool(Pool pool) {
+        return (this.pools.remove(pool.getId()) != null);
+    }
+
+    public ActivationKeyData setPools(Collection<AKPoolData> pools) {
+        this.pools.clear();
+
+        for (AKPoolData pool : pools) {
+            this.pools.put(pool.getPoolId(), pool);
+        }
+
+        return this;
+    }
+
+    public Collection<AKPoolData> getPools() {
+        return this.pools.values();
+    }
+
+    public boolean addProduct(Product product) {
+        String productId = product.getId();
+        this.products.put(productId, new AKProductData(productId));
+
+        return true;
+    }
+
+    public boolean removeProduct(Product product) {
+        return (this.products.remove(product.getId()) != null);
+    }
+
+    public ActivationKeyData setProducts(Collection<AKProductData> products) {
+        this.products.clear();
+
+        for (AKProductData product : products) {
+            this.products.put(product.getProductId(), product);
+        }
+
+        return this;
+    }
+
+    public Collection<AKProductData> getProducts() {
+        return this.products.values();
+    }
+
+    public boolean addContentOverride(ContentOverride override) {
+        return this.contentOverrides.add(new AKContentOverride(
+            override.getContentLabel(), override.getName(), override.getValue()
+        ));
+    }
+
+    public boolean removeContentOverride(ContentOverride override) {
+        List<AKContentOverride> remove = new LinkedList<AKContentOverride>();
+
+        for (AKContentOverride candidate : this.contentOverrides) {
+            if (candidate.getContentLabel().equals(override.getContentLabel()) &&
+                candidate.getName().equals(override.getName())) {
+                remove.add(candidate);
+            }
+        }
+
+        return this.contentOverrides.removeAll(remove);
+    }
+
+    public ActivationKeyData setContentOverrides(Collection<AKContentOverride> overrides) {
+        this.contentOverrides.clear();
+
+        if (overrides != null) {
+            this.contentOverrides.addAll(overrides);
+        }
+
+        return this;
+    }
+
+    public Collection<AKContentOverride> getContentOverrides() {
+        return this.contentOverrides;
```

Comment:
Yeah... only reason I didn't was because it would due to laziness, but you're right and we should probably just do it cleanly to begin with.

I'll make the change and throw in some documentation.